### PR TITLE
Argv

### PR DESCRIPTION
**Pull request by briancohan**
capture exception rather than parsing stderr

### CODE REVIEW

**Comment by briancohan** [comment]
Props to @mustafaquraish for the ideas on how to solve this (just wanted to put this in the record)

File: wtpython/__main__.py
Comment (mustafaquraish) [inline wtpython/__main__.py:36]: `argv[0]` should be the name of the script I believe, not `python`. You should just be able to do `sys.argv = args`.

This does seem a bit hacky, so maybe let's just replace `argv` right before we run the script with `runpy`, and then restore the original value right after.

File: wtpython/__main__.py
Comment (mustafaquraish) [inline wtpython/__main__.py:44]: Perhaps here would be a nice place to replace `sys.argv`, and restore it after (before returning the exception)

File: wtpython/__main__.py
Comment (mustafaquraish) [inline wtpython/__main__.py:50]: Have you removed the check to see if `exc` is None? We can exit early in this case. No point trying to call the StackOverflow API

File: wtpython/__main__.py
Comment (mustafaquraish) [inline wtpython/__main__.py:43]: Is this line necessary? We can just do

```py
try:
    runpy.run_path(args[0], run_name='__main__')
    return
except Exception as e:
    exc = e
    error = ''.join(traceback.format_exception_only(type(e), e)).strip()
```
This would solve one of my comments below as well.

File: wtpython/display.py
Comment (mustafaquraish) [inline wtpython/display.py:21]: I know the storing in module stuff was there earlier as well, but this seems really hacky storing stuff in globals inside the `display.py` file. If we're doing it this way, I think it would be nicer to have the globals in `core.py`, or better just a separate file which only stores this data. That way we don't have globals floating around in the files where we're actually doing the important things

File: wtpython/display.py
Comment (mustafaquraish) [inline wtpython/display.py:150]: maybe split it up into 2 lines? this is getting a bit spaghetti territory

File: wtpython/display.py
Comment (mustafaquraish) [inline wtpython/display.py:156]: ...It seems we're doing this a lot. Maybe just compute this once when we get the exception and then just monkey-patch the `Exception` object with this message?
```py
exc.display_message = ''.join(traceback.format_exception_only(type(exc), exc)).strip()
```

File: wtpython/__main__.py
Comment (briancohan) [reply wtpython/__main__.py:36]: it works either way, but you're right `sys.argv = args` is a bit cleaner.

File: wtpython/__main__.py
Comment (briancohan) [reply wtpython/__main__.py:44]: recreated a function for running that will alter `sys.argv` then reset in a `finally` block

File: wtpython/__main__.py
Comment (briancohan) [reply wtpython/__main__.py:50]: I had that in there, not sure where it went, but it's back now.

File: wtpython/display.py
Comment (briancohan) [reply wtpython/display.py:21]: before i stored them in display, i tried storing them in setttings.py. I had issues where the values were'nt keeping, so I moved it to display since that is the only module that should use the values. Ideally they would be passed to `Display` directly.

File: wtpython/display.py
Comment (briancohan) [reply wtpython/display.py]: I recognize that i have this once in main and twice in display, but I'm not sure it's quite worth creating a new function. I'm gonna leave it like it is for now, but we can always refactor it a bit later.

**Review by VagishVela — CHANGES_REQUESTED** [review]
Something weird is happening. It fails to identify `wtpython` if you do a fresh install using `venv` as in the README.md.

```
(template) ➜  Perceptive-Porcupines-Code-Jam-8 git:(argv) wtpython example/runs_with_error.py
zsh: command not found: wtpython
```

**Review by VagishVela — APPROVED** [review]
Looks good to me, do we need additional reviewers (e.g. @mustafaquraish ) or shall we merge?

**Comment by briancohan** [comment]
> Looks good to me, do we need additional reviewers (e.g. @mustafaquraish ) or shall we merge?

Feel free to merge if you think it's right, but I might wait for more reviews since it's the core of our app.

**Comment by mustafaquraish** [comment]
Haven't pulled and tested this, but the code looks fine to me. We can merge this if 3verythings working.

**Review by mustafaquraish — APPROVED** [review]

**Pull request merged**
